src/firefly/js/ui/ImageSelect.jsx

```diff
@@ -351,11 +353,37 @@ const toFilterSummary = (master, key, desc) => Object.entries(countBy(master, (d
 /*--------------------------- Data Product List ---------------------------------------*/
 
 function DataProductList({filteredImageData, groupKey, multiSelect}) {
+    // Create a field for storing default images in rgb field group
+    const {fireValueChange: fireDefaultImagesChange} = useFieldGroupConnector({fieldKey: 'defaultImages', groupKey: FG_KEYS.rgb});
+    const [getDefaultImages, ] = useFieldGroupValue('defaultImages', FG_KEYS.rgb);
+
+    const updateDefaultImages = (color) => ([coloredImageVal]) => {
+        // if it's a colored (red, green, or blue) DataProductList and if image for that color is selected for the first time
+        if (groupKey===FG_KEYS[color] && (coloredImageVal && !getDefaultImages()?.[color])) {
+            const selectedColoredImage = filteredImageData.find((d) => d.imageId === coloredImageVal);
+
+            // if color field (default images for rgb) is defined and default image specified for that color is same as selected image
+            if (selectedColoredImage?.color && selectedColoredImage.color[color]===coloredImageVal) {
+                fireDefaultImagesChange({value: selectedColoredImage.color, valid: true});
```

Comment:
If you take my suggestion above then.
```
 setDefaultImages(selectedColoredImage.color, {valid:true});
```
I am not sure `valid: true` is necessary but it might be cleaner to specify it.

Reply:
Yes valid: true is passed by default in setter so we can omit it.